contributor-docs/deprecating-components.md

```diff
@@ -0,0 +1,78 @@
+As we work on the maturity of our components, we will sometimes need to build components in a parallel track/bundle without breaking existing components. Eventually, the new components would replace the old ones in the main bundle and this document explains the process around it.
+
+## Developing a draft component
+
+Start building the new component outside of the main bundle. This includes the source code of the component being under the `src/drafts` folder and the component being exported in the draft bundle (`src/drafts/index.ts`). That way, folks who
```

Comment:
@siddharthkp instead of recommending these components go into `src/drafts`, do you think we should be encouraging folks to build directly in the new experimental components repo instead? https://github.com/github/experimental-react-components

Reply:
Oh super keen to hear about this 👀

Reply:
Yes! 

But given the change in direction for experimental-components, i'm not sure if it should be `github/experimental-react-components` or our own `primer/react-experimental`, another ADR? 😅

Reply:
Yes, probably. @justinbyo is going to be helping us figure out ownership models for shared React components looking forward.

In the meantime, are we comfortable shipping these docs with the direction to build direct in `experimental-react-components` for now, and we can always update if/when we make a change? We don't seem to deprecate components super often so I'm thinking that changing this guidance later wouldn't be a huge problem.

Reply:
If we are going to update here to build the components in `experiment-react-components`, some items (the ones related to the new component) in the [checklist](https://github.com/primer/react/pull/2733/files#diff-1760a697b5e00b53e2aecf5aefbec00b6fd0a8fbee6a228fa3e74adfd33ed458R22) won't be valid anymore and need to be updated to make sure that the "Experimental React components" are becoming "Primer" components. (just as an example we have our custom unit tests and storybook format and with the new docs update a json file that has metadata of the components). I am just not sure if deprecating component docs is the best place to write about "Primer" components. I  _think_ it deserves its own documentation -  like "what are Primer" component standards - and we can link here?

Reply:
Good point @broccolinisoup. In most cases we should have similar standards for components (including unit tests and Storybook) in the `experimental-react-components` repo as we do in `primer/react`. I'm admittedly not as familiar with all of the items in the checklist and whether they would apply exactly the same in `experimental-react-components`; what do you think @siddharthkp?

Reply:
> In most cases we should have similar standards for components (including unit tests and Storybook) in the experimental-react-components repo as we do in primer/react

I am a fan of this! 

I wasn't sure how much we can "influence" `experimental-react-components` to follow our Primer component standards. 
From my observation, `experimental-react-components` differ from PRC in the way of [documenting components](https://github.com/github/experimental-react-components/blob/main/docs/documenting-components.md) and they don't include some [util unit test functions](https://github.com/primer/react/blob/main/src/utils/testing.tsx) as PRC do.   

Super keen to hear @siddharthkp's thoughts.

Reply:
> I wasn't sure how much we can "influence" experimental-react-components to follow our Primer component standards.

Oooh, we definitely should influence it a lot. We can even assist (and enforce) practices by baking them into the [react toolchain](https://github.com/github/experimental-react-components/tree/main/packages/%40github/react-toolchain).

However, we also keep the barrier for contribution pretty low. Experimental components are meant to be [pre-alpha in the lifecycle](https://primer.style/contribute/component-lifecycle/), more context in [ADR 008](https://github.com/primer/react/blob/main/contributor-docs/adrs/adr-008-experimental-components.md)

> From my observation, experimental-react-components differ from PRC in the way of [documenting components](https://github.com/github/experimental-react-components/blob/main/docs/documenting-components.md) and they don't include some [util unit test functions](https://github.com/primer/react/blob/main/src/utils/testing.tsx) as PRC do.

Some of this, like the docs, is intentional, we chose storybook docs over doctocat to make it easier (and less elaborate) to document than primer. The lack of test functions isn't part of the strategy, that's just because we've only had one iteration of this till now :)

Reply:
>Oooh, we definitely should influence it a lot. We can even assist (and enforce) practices by baking them into the [react toolchain](https://github.com/github/experimental-react-components/tree/main/packages/%40github/react-toolchain).

Awesome to hear! 

>However, we also keep the barrier for contribution pretty low. Experimental components are meant to be [pre-alpha in the lifecycle](https://primer.style/contribute/component-lifecycle/), more context in [ADR 008](https://github.com/primer/react/blob/main/contributor-docs/adrs/adr-008-experimental-components.md)

This is a pretty good point!

>Some of this, like the docs, is intentional, we chose storybook docs over doctocat to make it easier (and less elaborate) to document than primer. The lack of test functions isn't part of the strategy, that's just because we've only had one iteration of this till now :)

Gotcha! 

It feels to me that moving experimental components from `react-experimental-components` is a slightly different process than deprecating components, especially around the practises.

Another question - if an existing component was to be re-written by a Primer engineer (following all Primer standards), would we still start in the `experimental-react-components` repo or in our own repos?

cc @lesliecdubs

Reply:
Hello! @siddharthkp and @lesliecdubs

Seems like we still have questions regarding to `experimental-react-components` repo's practises and how they will make their way to Primer. Would it be okay to go ahead with this section as is (creating the components in the `drafts` folder) until we have a clear practise for `experimental-react-components`?  At least we will treat this document as the current practise and we can update as we go?